src/lib.rs

```diff
@@ -63,34 +62,36 @@ impl ToOperation for TodoOperation {
     }
 }
 
-pub fn operation_from_string(s: String) -> TodoOperation {
-    let vec: Vec<&str> = s.split('\0').collect();
-
-    TodoOperation {
-        operation: match vec[0] {
-            "add" => OperationType::Add,
-            "remove" => OperationType::Remove,
-            _ => panic!("expected valid operation type, got {}", vec[0]),
-        },
-        todo: TodoData {
-            engine: vec[1].to_string(),
-            rule_id: vec[2].to_string(),
-            file_path: vec[3].to_string(),
-            range: Range {
-                start: Position {
-                    line: i32::from_str_radix(vec[4], 10).expect("valid range.start.line"),
-                    column: i32::from_str_radix(vec[5], 10).expect("valid range.start.column"),
-                },
-                end: Position {
-                    line: i32::from_str_radix(vec[6], 10).expect("valid range.end.line"),
-                    column: i32::from_str_radix(vec[7], 10).expect("valid range.end.column"),
+impl From<&str> for TodoOperation {
```

Comment:
Implementing `From` here is the idiomatic way to do this. It also takes advantage of the fact that if you `impl From<SomeType> for SomeOtherType` you can always do `instanceOfSomeOtherType.into()` to get a `SomeType` (see below).